src/PrimitiveObsession.Test/EngineContainerBuilderTest.cs

```diff
@@ -18,47 +19,49 @@ public EngineContainerBuilderTest()
             var builder = new ContainerBuilder();
             builder.RegisterModule(sut);
             _container = builder.Build();
+            _scope = _container.BeginLifetimeScope();
```

Comment:
I componenti non vanno mai risolti dal container. Ho corretto sia qui nel test che nel `Program`.

La raccomandazione e la spiegazione le trovi qui http://docs.autofac.org/en/latest/lifetime/

It is important to always resolve services from a lifetime scope and not the root container. Due to the disposal tracking nature of lifetime scopes, if you resolve a lot of disposable components from the container (the “root lifetime scope”), you may inadvertently cause yourself a memory leak. The root container will hold references to those disposable components for as long as it lives (usually the lifetime of the application) so it can dispose of them. You can change disposal behavior if you choose, but it’s a good practice to only resolve from a scope. If Autofac detects usage of a singleton or shared component, it will automatically place it in the appropriate tracking scope.